File: STUDENT-HUB-ANDROID-NATIVE-FCM-UPDATED(2)/STUDENT-HUB-ANDROID-NATIVE-FCM-UPDATED/app/src/main/java/com/studenthub/app/util/Slug.java
package com.studenthub.app.util;

/** Same slugify logic as the web app's slugify() in app.js (e.g. "Class 9" -> "class-9"). */
public class Slug {
    public static String slugify(String input) {
        if (input == null) return "";
        return input.trim().toLowerCase()
                .replaceAll("[^a-z0-9]+", "-")
                .replaceAll("^-+|-+$", "");
    }
}

File: android/app/src/main/java/com/studenthub/app/util/Slug.kt
package com.studenthub.app.util

object Slug {
    fun slugify(input: String): String {
        return input.lowercase().replace(Regex("[^a-z0-9]+"), "-").trim('-')
    }

    fun classRoomId(classLevel: String): String {
        return "class-" + slugify(classLevel)
    }
}
